src/snowflake/cli/plugins/spcs/image_repository/manager.py

```diff
@@ -51,8 +51,21 @@ def get_repository_api_url(self, repo_url):
 
         return f"{scheme}://{host}/v2{path}"
 
-    def create(self, name: str):
+    def create(
+        self,
+        name: str,
+        if_not_exists: bool,
+        replace: bool,
+    ):
+
+        create_statement = (
+            f"{'create or replace' if replace else 'create'} image repository"
+        )
+        if if_not_exists:
+            create_statement = f"{create_statement} if not exists"
```

Comment:
Is it possible to pass in both `--replace` and `--if-not-exists` right now? This is a bit of a drive-by review and I do not have time to check, but we should enforce maximum one of the three introduced `OverrideableOptions` if possible.

Reply:
(otherwise this will generate invalid SQL)

Reply:
good point, as far as I know IF EXISTS, IF NOT EXISTS, and OR REPLACE should be mutually exclusive for all commands so I'll add a callback to check that at most one is set to true.